packages/core/src/external/cda/process-attachments.ts

```diff
@@ -0,0 +1,219 @@
+import {
+  Attachment,
+  Bundle,
+  CodeableConcept,
+  Coding,
+  DocumentReference,
+  Extension,
+  Identifier,
+} from "@medplum/fhirtypes";
+import { executeWithNetworkRetries } from "@metriport/shared";
+import { buildDayjs } from "@metriport/shared/common/date";
+import { cloneDeep } from "lodash";
+import { MedicalDataSource, isMedicalDataSource } from "..";
+import { createAttachmentUploadPath } from "../../domain/document/upload";
+import { createFilePathFromFileName } from "../../domain/filename";
+import { executeAsynchronously } from "../../util/concurrency";
+import { Config } from "../../util/config";
+import { out } from "../../util/log";
+import { sizeInBytes } from "../../util/string";
+import { uuidv4 } from "../../util/uuid-v7";
+import { S3Utils, UploadParams } from "../aws/s3";
+import { cqExtension } from "../carequality/extension";
+import { cwExtension } from "../commonwell/extension";
+import { FhirClient } from "../fhir/api/api";
+import { convertCollectionBundleToTransactionBundle } from "../fhir/bundle/convert-to-transaction-bundle";
+import { buildDocIdFhirExtension } from "../fhir/shared/extensions/doc-id-extension";
+
+const s3Utils = new S3Utils(Config.getAWSRegion());
+const s3BucketName = Config.getMedicalDocumentsBucketName();
+
+export async function processAttachments({
+  b64Attachments,
+  cxId,
+  patientId,
+  fileName,
+  medicalDataSource,
+  fhirApi,
+}: {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  b64Attachments: any[];
+  cxId: string;
+  patientId: string;
+  fileName: string;
+  medicalDataSource?: string | undefined;
+  fhirApi?: FhirClient;
+}) {
+  const { log } = out(`processAttachments - cxId ${cxId}, patientId: ${patientId}`);
+  const filePath = createFilePathFromFileName(fileName);
```

Comment:
likely we are saving all these under the path of the original file right? might be easy to simply send the original full path, and simply make filePath = `fullPath.split('/')[0]` or something.

Reply:
already fixed